client_options.go

```diff
@@ -3,68 +3,105 @@ package walletclient
+	"github.com/bitcoinschema/go-bitcoin/v2"
+	"github.com/libsv/go-bk/bec"
+	"github.com/libsv/go-bk/wif"
+	"github.com/pkg/errors"
 )
 
-// WithXPriv will set xPrivString on the client
-func WithXPriv(xPrivString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPrivString = xPrivString
-		}
-	}
+// WalletClientConfigurator is the interface for configuring WalletClient
+type WalletClientConfigurator interface {
+	Configure(c *WalletClient)
 }
 
-// WithXPub will set xPubString on the client
-func WithXPub(xPubString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.xPubString = xPubString
-		}
+// XPriv sets the xPrivString field of a WalletClient
+type XPriv struct {
+	XPrivString *string
+}
+
+func (w *XPriv) Configure(c *WalletClient) {
+	var err error
+	if c.xPriv, err = bitcoin.GenerateHDKeyFromString(*w.XPrivString); err != nil {
+		c.xPriv = nil
 	}
 }
 
-// WithAccessKey will set the access key on the client
-func WithAccessKey(accessKeyString string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.accessKeyString = accessKeyString
-		}
+// XPub sets the xPubString on the client
+type XPub struct {
+	XPubString *string
+}
+
+func (w *XPub) Configure(c *WalletClient) {
+	var err error
+	if c.xPub, err = bitcoin.GetHDKeyFromExtendedPublicKey(*w.XPubString); err != nil {
+		w.XPubString = nil
 	}
+
 }
 
-// WithHTTP will overwrite the default client with a custom client
-func WithHTTP(serverURL string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTP(serverURL))
-		}
+// AccessKey sets the accessKeyString on the client
+type AccessKey struct {
+	AccessKeyString *string
+}
+
+func (w *AccessKey) Configure(c *WalletClient) {
+	var err error
+	if c.accessKey, err = w.initializeAccessKey(); err != nil {
+		c.accessKey = nil
 	}
 }
 
-// WithHTTPClient will overwrite the default client with a custom client
-func WithHTTPClient(serverURL string, httpClient *http.Client) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithHTTPClient(serverURL, httpClient))
-		}
+// AdminKey sets the admin key for creating new xpubs
+type AdminKey struct {
+	AdminKeyString *string
+}
+
+func (w *AdminKey) Configure(c *WalletClient) {
+	var err error
+	c.adminXPriv, err = bitcoin.GenerateHDKeyFromString(*w.AdminKeyString)
+	if err != nil {
+		c.adminXPriv = nil
 	}
 }
 
-// WithAdminKey will set the admin key for admin requests
-func WithAdminKey(adminKey string) ClientOps {
-	return func(c *WalletClient) {
-		if c != nil {
-			c.transportOptions = append(c.transportOptions, transports.WithAdminKey(adminKey))
-		}
+// HTTP sets the URL and HTTP client of a WalletClient
+type HTTP struct {
+	ServerURL  *string
+	HTTPClient *http.Client
+}
+
+func (w *HTTP) Configure(c *WalletClient) {
+	c.server = *w.ServerURL
```

Comment:
It would be awesome if we would only require to pass host part.
What I mean. currently user/adopter needs to provide server url in a form like this:
`https://localhost:3003/v1`, 
IMHO it would be much better for both us and them to require only:
`https://localhost:3003` and add the `v1` path part for example here.
This will prevent mistakes on the user side, and will open a possibility to change this path (for example to v2) in the future.

Reply:
Nice idea 👍 However, we should remember to implement it to the `js-client` also. Maybe additional task? WDYT @dorzepowski ?

Reply:
well not so sure if this is great idea what if we will have v2 ?

Reply:
ok its implemented .. i will fix tests and will push it